### DISCUSSION TITLE
Select.clear() doesn't trigger Select.Changed event

### DISCUSSION THREAD

**Discussion by Tim-Cao** [Textualize/textual#3849] [Q&A]
``` python
from textual.app import App, ComposeResult
from textual.containers import *
from textual.widgets import *
from textual import work


class ScriptApp(App):
    CSS = """
.unselected {
    display: none;
}

.unselected.visible {
    display: block;
}
"""

    def compose(self) -> ComposeResult:
        PROJECT_KEY = [
            ("LPS", "LPS"),
            ("LRQA", "LRQA"),
            ("LRAC", "LRAC"),
            ("COMMERCE", "COMMERCE"),
        ]

        yield Header()
        yield Label("Project Key:", id="project-key-label-2")
        yield Select(PROJECT_KEY, id="project-key-2")
        yield Label("Summary:", id="summary-label", classes="unselected")
        yield Input(id="summary", classes="unselected")
        yield Static()
        yield Button("Submit", id="button-7")
        yield Footer()

    @work(exclusive=True, thread=True)
    def create_issue(self) -> None:
        self.query_one("#button-7").disabled = True

        self.query_one("#summary").value = ""
        self.query_one("#project-key-2").clear()
        self.query_one("#button-7").disabled = False

    def on_button_pressed(self, event: Button.Pressed) -> None:
        if event.button.id == "button-7":
            self.create_issue()

    def on_select_changed(self, event: Select.Changed) -> None:
        self.query_one("#summary").set_class(event.value != Select.BLANK, "visible")
        self.query_one("#summary-label").set_class(
            event.value != Select.BLANK, "visible"
        )


app = ScriptApp()

if __name__ == "__main__":
    app.run()
```

I would like to hide the _**Summary**_ input field after submit since the selection of **_Project Key_** will be cleared.

**Comment by TomJGooding** [comment]
I've only skimmed your code, but for starters I think you should observe the warning from [the documentation on threaded workers](https://textual.textualize.io/guide/workers/#thread-workers) about calling methods on your UI directly,

**Reply by TomJGooding** [reply]
Sorry also just to add, please use syntax highlighting for code examples. See the pinned discussion on [The Art of the Minimal Reproducible Example](https://github.com/Textualize/textual/discussions/3361#discussion-5652292)

**Reply by Tim-Cao** [reply]
Thanks for your reminding. I've added the syntax highlight. However, I still didn't figure out how to make it work with call from thread.
@TomJGooding, would you mind investigating deeply?

**Reply by TomJGooding** [reply]
Thanks, it just makes the code easier to read.

After looking at this again, you're right that "Select.clear() doesn't trigger Select.Changed event". I'd forgotten that this has come up before:

- #3417

Until there's an answer on this, I think a simple workaround for your application would be to remove the class directly after your call to `clear`.

**Reply by Tim-Cao** [reply]
Thanks for your workaround. Will you have a plan to fix this issue recently? @TomJGooding

**Reply by TomJGooding** [reply]
I'm not a maintainer of Textual so I don't get the final say on whether this is the correct behaviour I'm afraid!

**Reply by davep** [reply]
Just to update this, see also #3869.
